ecommerce/backend/API/views/product_list.py

```diff
@@ -0,0 +1,43 @@
+from django.http.response import HttpResponseForbidden, HttpResponseNotAllowed
+from rest_framework import exceptions
+    
+from rest_framework.response import Response
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework import status
+from rest_framework.parsers import JSONParser
+
+from ..utils import permissions, Role
+from ..models import User, ShoppingCartItem, Product, Address, Order, Purchase, Card, ProductList, ProductListItem
+from ..serializers.product_list_serializer import ProductListResponseSerializer
+from ..utils import authentication, order_status
+
+
+@api_view(['POST', 'GET'])
+@permission_classes([permissions.AllowAnonymous])
+def product_list_create(request):
+    jwt = authentication.JWTAuthentication()
+    user = jwt.authenticate(request=request)[0]
+    
+    if request.method == "POST":
+        list_name = request.data.get("name")
+
+        product_list = ProductList(user_id=user.pk, name=list_name)
+        product_list.save()
+
+        return Response({'id': product_list.pk, 'status': { 'successful': True, 'message': "List is successfully created."}})
+
+    elif request.method == "GET":
+        response_product_list = []
+        product_lists = ProductList.objects.filter(user_id=user.pk)
+
```

Comment:
Check the edge case if no such product_list exists.

```suggestion
if not product_lists:
    return Response({'status': {'successful': False, 'message':'No such list exists.'})

```

Reply:
This is not a query function, this retrieves all lists of a user by user's id. So, if user has no lists created, it simply returns an empty list. Edge case may not be a problem.

Reply:
I think Umut is right. It is not a problem. Frontend may check if the list is empty and then they can print something like "you have no list".

Reply:
Actually, I traverse all product_lists with for loop, so if no item in it, response will return empty